design_docs/timekeeping_reorg.md

```diff
@@ -0,0 +1,185 @@
+        - DDD SSSSS
+        - hh.mm.ss
+        - hh:mm:ss
+        - YYYY-MM-DD
+        - YYYY-MM
+        - SSSSS
+
+        Note: either underscores or spaces can be used to separate the date
+        from the time portion of the string.
+
+    Returns
+    -------
+    datetime : A `datetime.datetime` object
+
+    Raises
+    ------
+    ValueError
+        If an invalid `dateString` is supplied.
+
+    Author
+    ------
+    Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/04/2017
+    """
+```
+
+As long as `relativedelta` objects rather than `datetime.timedelta` objects are used to increment
+`datetime.datetime` objects, `datetime.datetime` can be used to represent dates on either the Gregorian
+or the 365-day calendar.
+
+<h2> Implementation: Capability of incrementing dates by a number of years and/or months <br>
+Date last modified: 2017/02/06 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+The implementation is on the branch:
+https://github.com/xylar/MPAS-Analysis/tree/timekeeping_reorg
+and in PR #102
+
+The proposed implementation uses `dateutil.relativedelta.relativedelta` to compute the expected 
+increments in years and months (as well as days, hours, minutes and seconds, as needed).
+The function for converting a date string to a `relativedelta` is documented as follows:
+```python
+from dateutil.relativedelta import relativedelta
+...
+def stringToRelativedelta(dateString, calendar='gregorian'):
+    """
+    Given a date string and a calendar, returns an instance of
+    `dateutil.relativedelta.relativedelta`
+
+    Parameters
+    ----------
+    dateString : string
+        A date and time in one of the following formats:
+        - YYYY-MM-DD hh:mm:ss
+        - YYYY-MM-DD hh.mm.ss
+        - YYYY-MM-DD SSSSS
+        - DDD hh:mm:ss
+        - DDD hh.mm.ss
+        - DDD SSSSS
+        - hh.mm.ss
+        - hh:mm:ss
+        - YYYY-MM-DD
+        - YYYY-MM
+        - SSSSS
+
+        Note: either underscores or spaces can be used to separate the date
+        from the time portion of the string.
+
+    calendar: {'gregorian', 'gregorian_noleap'}, optional
+        The name of one of the calendars supported by MPAS cores
+
+    Returns
+    -------
+    relativedelta : A `dateutil.relativedelta.relativedelta` object
+
+    Raises
+    ------
+    ValueError
+        If an invalid `dateString` is supplied.
+
+    Author
+    ------
+    Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/04/2017
+    """
+```
+
+<h1> Testing </h1>
+
+<h2> Testing and Validation: Date string parsing supports both MPAS calendars <br>
+Date last modified: 2017/02/08 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+Analysis will be run on Edison with all available configurations found in `configs/edison`.  As there
+are currently no plans to run with the `gregorian` calendar option, we do not have test runs that use this
+calendar.  If this situation changes in the future, we'll test at that time.
```

Comment:
Shouldn't we still test functionality via py.test, even if it isn't fully in use?  Otherwise, we could argue we don't need to fully flesh out the option that is neither tested nor used.